docs/src/prototypes/CopyToClipboard/CopyToClipboard.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react'
+import { Popup, Button, Text, Portal, Ref } from '@stardust-ui/react/src'
+import * as copyToClipboard from 'copy-to-clipboard'
+
+export type CopyToClipboardProps = {
+  value: string
+  copyPrompt: string
+  timeout?: number
+  attached?: boolean
+  pointing?: boolean
+}
+
+class CopyToClipboard extends React.Component<CopyToClipboardProps> {
+  state = {
+    copied: false,
+    copiedHeight: 0,
+    copiedWidth: 0,
+  }
+  timeoutId = undefined
+  copiedTextRef = React.createRef<HTMLElement>()
+
+  static defaultProps = {
+    timeout: 4000,
+    attached: false,
+    pointing: false,
+  }
+
+  handleClick = () => {
+    if (this.timeoutId !== undefined) {
+      clearTimeout(this.timeoutId)
+    }
+    const { timeout, value } = this.props
+    this.setState({
+      copied: true,
+    })
+    this.timeoutId = setTimeout(() => {
+      this.setState({ copied: false })
+    }, timeout)
+    copyToClipboard(value)
+  }
+
+  componentDidMount() {
+    if (this.copiedTextRef.current === null) {
+      return
+    }
+    this.setState({
+      copiedHeight: this.copiedTextRef.current.clientHeight,
+      copiedWidth: this.copiedTextRef.current.clientWidth,
+    })
+  }
+
+  render() {
+    const { attached, pointing, copyPrompt } = this.props
+    const hideContent = this.state.copied && !attached
+    const copiedStr = 'Copied to clipboard'
+    const copiedText = hideContent ? (
+      <Text size="larger" styles={{ padding: '5px 5px 5px 5px' }}>
+        {copiedStr}
+      </Text>
+    ) : (
+      <Text>{copiedStr}</Text>
+    )
+    const copyText = <Text>{copyPrompt}</Text>
+    let popupContent = this.state.copied ? copiedText : copyText
+    if (hideContent) {
+      popupContent = <Text />
+    }
+    const button = (
+      <Button
+        icon="clipboard-copied-to"
+        iconOnly
+        onClick={this.handleClick}
+        variables={siteVariables => ({
+          colorHover: !this.state.copied && siteVariables.colors.brand[400],
+        })}
+      />
+    )
+    const hiddenPopupVariables = {
+      contentBackgroundColor: 'transparent',
+    }
+    const hiddenPopupContentVariables = {
+      boxShadow: 'none',
+      borderColor: 'transparent',
+    }
+    const portalContentStyle = {
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      position: 'fixed' as 'fixed',
+      backgroundColor: 'black',
+      color: 'white',
+      zIndex: 1000,
+      top: `calc(50% - ${this.state.copiedHeight / 2}px)`,
+      left: `calc(50% - ${this.state.copiedWidth / 2}px)`,
+    }
+    return (
+      <>
+        <Popup
+          pointing={pointing}
+          position="below"
+          align="center"
+          trigger={button}
+          content={{
+            content: popupContent,
+            variables: {
+              ...(hideContent && hiddenPopupContentVariables),
+            },
+          }}
+          on={['hover', 'context']}
```

Comment:
I tried the prototype - it shows the message correctly after click, but then when I move my cursor outside of the button and return back to it, it shows the message again - I believe it's because of 'hover'. I think it should be on={[]}. Not sure if Toolbar would behave better or worse

Reply:
This is intentional. Upon clicking the button, the components gets into a _copied_ state. During the duration of this state, I'm showing 'Copied to clipboard' message. Would you prefer to show the original "Copy commit ID" message instead?

Reply:
the message should be visible also when you do not hover/focus on the button.

Reply:
It is now visible when the button is not hovered/focused